packages/cli/src/lib/setup/setupSetup.ts

```diff
@@ -182,164 +233,172 @@ Please DO NOT copy files manually into ioBroker storage directories!`
+
+            let existingObj: ioBroker.Object | undefined | null;
+            try {
+                existingObj = await this.objects.getObjectAsync(obj._id);
+            } catch {
+                // ignore
+            }
+            if (!existingObj || existingObj._id.startsWith('_design/')) {
+                obj.from = `system.host.${tools.getHostName()}.cli`;
+                obj.ts = Date.now();
+                await this.objects.setObjectAsync(obj._id, obj);
+                console.log(`object ${obj._id} ${!existingObj ? 'created' : 'updated'}`);
+                setTimeout(this.dbSetup, 25, iopkg, ignoreExisting, callback);
+            } else {
+                !ignoreExisting && console.log(`object ${obj._id} yet exists`);
+                setTimeout(this.dbSetup, 25, iopkg, ignoreExisting, callback);
+            }
         } else {
-            await tools.createUuid(objects);
+            await tools.createUuid(this.objects);
+            let configObj: ConfigObject | null | undefined;
             // check if encrypt secret exists
-            objects.getObject('system.config', (err, obj) => {
-                let configFixed = false;
-                if (obj && obj.type !== 'config') {
-                    obj.type = 'config';
-                    obj.from = `system.host.${tools.getHostName()}.cli`;
-                    obj.ts = Date.now();
-                    configFixed = true;
-                }
-                if (obj && (!obj.native || !obj.native.secret)) {
-                    require('crypto').randomBytes(24, (ex, buf) => {
-                        obj.native = obj.native || {};
-                        obj.native.secret = buf.toString('hex');
-                        obj.from = `system.host.${tools.getHostName()}.cli`;
-                        obj.ts = Date.now();
-                        objects.setObject('system.config', obj, () => setupReady(obj, callback));
-                    });
+            try {
+                configObj = await this.objects.getObject('system.config');
+            } catch {
+                // assume non existing
+            }
+
+            let configFixed = false;
+            if (configObj && configObj.type !== 'config') {
+                configObj.type = 'config';
+                configObj.from = `system.host.${tools.getHostName()}.cli`;
+                configObj.ts = Date.now();
+                configFixed = true;
+            }
+
+            if (configObj && (!configObj.native || !configObj.native.secret)) {
+                crypto.randomBytes(24, (ex, buf) => {
+                    configObj!.native = configObj!.native || {};
+                    configObj!.native.secret = buf.toString('hex');
+                    configObj!.from = `system.host.${tools.getHostName()}.cli`;
+                    configObj!.ts = Date.now();
+                    this.objects!.setObject('system.config', configObj!, () => this.setupReady(configObj!, callback));
+                });
```

Comment:
Just make it sync.
```suggestion
                const buf = crypto.randomBytes(24)
                configObj.native = configObj.native || {};
                configObj.native.secret = buf.toString('hex');
                configObj.from = `system.host.${tools.getHostName()}.cli`;
                configObj.ts = Date.now();
                this.objects!.setObject('system.config', configObj, () => this.setupReady(configObj, callback));
```
Although I think the !s won't be necessary if you define configObj as a const with the above style.